HHS/Head-Start-TTADP
# [Prod] Backend for filtering on goal name, UI changes to further fix where user is creator and approver

## Description of change

### 2725
Per user request - if a user is an approver and the creator, and the report is in "needs action", allow that user to reset the report to draft. Some updates to the UI are necessary in addition to the backend changes made yesterday.

### 2643, 2644, 2467

This change just puts the backend code in place for the goal name filter for activity reports, RTTAPA, and RTR search. There will be several of these PRs incoming; they are all prerequisite for the frontend ticket, which will come last.

## How to test

To test 2725:

Set yourself as an approver and creators on a report and then reset that report to draft. Remove yourself as an approver, choose a different approver, and resubmit. Confirm that the functionality doesn't exist for approvers who aren't creators, or creators who aren't approvers (on a given report)

2643, 2644, and 2647 cannot be tested through the UI

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2647
* https://jira.acf.gov/browse/TTAHUB-2644
* https://jira.acf.gov/browse/TTAHUB-2643
* https://jira.acf.gov/browse/TTAHUB-2725


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
